Modules/_asynciomodule.c

```diff
@@ -3345,9 +3371,16 @@ _asyncio__register_task_impl(PyObject *module, PyObject *task)
 /*[clinic end generated code: output=8672dadd69a7d4e2 input=21075aaea14dfbad]*/
 {
     asyncio_state *state = get_asyncio_state(module);
-    if (register_task(state, task) < 0) {
-        return NULL;
+    if (!Task_Check(state, task)) {
+        PyObject *res = PyObject_CallMethodOneArg(state->scheduled_tasks,
+                                                  &_Py_ID(add), task);
+        if (res == NULL) {
+            return NULL;
+        }
+        Py_DECREF(res);
+        Py_RETURN_NONE;
     }
+    register_task(state, (TaskObj *)task);
```

Comment:
assuming the common case is "native asyncio task", wouldn't it be more efficient (in terms of compiler branch prediction or whatnot) to have the `register_task` branch inside the `if`?

```suggestion
    if (Task_Check(state, task)) {
        register_task(state, (TaskObj *)task);
        Py_RETURN_NONE;
    }
    PyObject *res = PyObject_CallMethodOneArg(state->scheduled_tasks,
                                              &_Py_ID(add), task);
    if (res == NULL) {
        return NULL;
    }
    Py_DECREF(res);
```

Reply:
I think PGO will reorder this anyways so I wouldn't worry about it.

Reply:
On second thought I'll just change the order as you suggest, thanks. Windows already struggles with PGO so let's not make it worse.